sql/parse_tree_helpers.cc

```diff
@@ -358,12 +359,14 @@ bool sp_create_assignment_instr(THD *thd, const char *expr_end_ptr) {
   }
 
   /* Remember option_type of the currently parsed LEX. */
-  enum_var_type inner_option_type = lex->option_type;
+  const enum_var_type inner_option_type = lex->option_type;
```

Comment:
Not too important, but I wonder if this code path actually works? It'd be nice to have simple test for set statements in stored procedures.

Reply:
It doesn't work because of an upstream bug:
https://bugs.mysql.com/bug.php?id=95734